Node/src/botbuilder-teams.d.ts

```diff
@@ -213,62 +195,593 @@ export interface O365ConnectorCardActionBase {
+ * 
+ * @member {string} [title] Input title that will be shown as the placeholder
+ * 
+ * @member {string} [value] Default value for this input field
+ */
+export interface IO365ConnectorCardInputBase {
+  id: string;
+  isRequired?: boolean;
+  title: string;
+  value: string;
+}
+
+export interface IIsO365ConnectorCardInputBase {
+  toInput(): IO365ConnectorCardInputBase;
+}
+
+/**
+ * @interface
+ * Interface of O365 connector card text input
+ * 
+ * @member {boolean} [isMultiline] Define if text input is allowed for multiple lines. Default value is false.
+ * 
+ * @member {number} [maxLength] Maximum length of text input. Default value is unlimited.
+ *  
+ */
+export interface IO365ConnectorCardTextInput extends IO365ConnectorCardInputBase{
+  isMultiline?: boolean;
+  maxLength?: number;
+}
+
+/**
+ * @interface
+ * Interface of O365 connector card date input
+ * 
+ * @member {boolean} [includeTime] Include time input field. Default value  is false (date only).
+ *  
+ */
+export interface IO365ConnectorCardDateInput extends IO365ConnectorCardInputBase{
+  includeTime?: boolean;
+}
+
+/**
+ * @interface
+ * Interface of O365 connector card multiple choice input
+ * 
+ * @member {array} [choices] Set of choices whose each item can be in any subtype of IO365ConnectorCardMultichoiceInputChoice.
+ * 
+ * @member {IO365ConnectorCardMultichoiceInputStyle} [style] Choice item rendering style. Could be 'compact' (default) or 'expanded'.
+ * 
+ * @member {boolean} [isMultiSelect] Define if this input field allows multiple selections. Default value is false.
+ *  
+ */
+export interface IO365ConnectorCardMultichoiceInput extends IO365ConnectorCardInputBase{
+  choices: IO365ConnectorCardMultichoiceInputChoice[];
+  style?: IO365ConnectorCardMultichoiceInputStyle;
+  isMultiSelect?: boolean; 
+}
+
+/**
+ * @type
+ * Type of literal strings used for multi-choice input (IO365ConnectorCardMultichoiceInput) rendering style.
+ *
+ */
+export type IO365ConnectorCardMultichoiceInputStyle = 'compact' | 'expanded';
+
+/**
+ * @interface
+ * Interface of O365 connector card multiple choice input item
+ * 
+ * @member {string} [display] The text rednered on ActionCard.
+ * 
+ * @member {string} [value] The value received as results.
+ * 
+ */
+export interface IO365ConnectorCardMultichoiceInputChoice {
+  display: string;
+  value: string;
+}
+
+export interface IIsO365ConnectorCardMultichoiceInputChoice {
+  toChoice(): IO365ConnectorCardMultichoiceInputChoice;
+}
+
+/**
+ * @interface
+ * Interface of O365 connector card HttpPOST invoke query
+ * 
+ * @member {string} [body] The results of body string defined in IO365ConnectorCardHttpPOST with substituted input values
+ * 
+ * @member {string} [actionId] Action Id associated with the HttpPOST action button triggered, defined in IO365ConnectorCardActionBase.
+ *  
+ */
+export interface O365ConnectorCardActionQuery {
```

Comment:
@aosolis I'll revise this, but do you think we need to change it for compose extension parts? There're several interfaces for compose extensions.